lib/datadog/core/telemetry/client.rb

```diff
@@ -74,13 +94,48 @@ def client_configuration_change!(changes)
           @emitter.request('app-client-configuration-change', data: { changes: changes, origin: 'remote_config' })
         end
 
+        def add_count_metric(namespace, name, value, tags)
+          return if !@metrics_enabled || forked?
+
+          @metric_queue.add_metric(namespace, name, value, tags, Metric::Count)
+        end
+
+        def add_rate_metric(namespace, name, value, tags)
+          return if !@metrics_enabled || forked?
+
+          @metric_queue.add_metric(namespace, name, value, tags, Metric::Rate)
+        end
+
+        def add_gauge_metric(namespace, name, value, tags)
+          return if !@metrics_enabled || forked?
+
+          @metric_queue.add_metric(namespace, name, value, tags, Metric::Gauge)
+        end
+
+        def add_distribution_metric(namespace, name, value, tags)
+          return if !@metrics_enabled || forked?
+
+          @metric_queue.add_metric(namespace, name, value, tags, Metric::Distribution)
+        end
+
         private
 
         def heartbeat!
           return if !@enabled || forked?
 
           @emitter.request(:'app-heartbeat')
         end
+
+        def flush_metrics!
+          return if !@metrics_enabled || forked?
+
+          # Send metrics
+          @metric_queue.build_metrics_payload do |metric_type, payload|
+            @emitter.request(metric_type, payload: payload)
```

Comment:
sorry if I missed something obvious, but what if another metric arrives when this request is running and GIL is released? would this metric be lost when we call `@metric_queue = MetricQueue.new` afterwards?